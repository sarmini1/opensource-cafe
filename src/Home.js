import React from "react";
import { Card, CardBody, CardTitle } from "reactstrap";

/** Home 
 * 
 * Props:
 * - drinks: array of objects [{id, name, description, recipe, serve}...]
 * - snacks: array of objects [{id, name, description, recipe, serve}...]
 * 
 * State:
 * - none
 *  
 * App--> NavBar, Routes --> Home
 * */
function Home({ drinks, snacks }) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h3 className="font-weight-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </h3>
            <h4>
              Now offering a selection of {snacks.length} different
              snacks and {drinks.length} different drinks!
            </h4>
          </CardTitle>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;