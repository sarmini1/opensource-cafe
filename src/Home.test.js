import React from 'react';
import { render } from '@testing-library/react';
import Home from "./Home";

const testDrinks = [
  {
    "id": "martini",
    "name": "Martini",
    "description": "An ice-cold, refreshing classic.",
    "recipe": "Mix 3 parts vodka & 1 part dry vermouth.",
    "serve": "Serve very cold, straight up."
  },
  {
    "id": "negroni",
    "name": "Negroni",
    "description": "A nice drink for a late night conversation.",
    "recipe": "Mix equal parts of gin, Campari, and sweet vermouth.",
    "serve": "Serve cold, either on the rocks or straight up."
  }];

const testSnacks = [
  {
    "id": "nachos",
    "name": "Nachos",
    "description": "An American classic!",
    "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
    "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
  },
  {
    "id": "hummus",
    "name": "Hummus",
    "description": "Sure to impress your vegan friends!",
    "recipe": "Purchase one container of hummus.",
    "serve": "Place unceremoniously on the table, along with pita bread."
  }];


it('renders without crashing', function () {
  render(<Home drinks={testDrinks} snacks={testSnacks} />);
});

it("matches snapshot", function () {
  const { container } = render(<Home
    drinks={testDrinks}
    snacks={testSnacks} />);
  expect(container).toMatchSnapshot();
});


it('displays menu of drink options when rendered', function () {
  const { debug, container } = render(<Home
    drinks={testDrinks}
    snacks={testSnacks} />);

  debug(container);

  expect(container).toContainHTML("2 different snacks");
  expect(container).toContainHTML("2 different drinks");
});