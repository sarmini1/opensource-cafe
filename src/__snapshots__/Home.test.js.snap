// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    class="col-md-8"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <div
          class="card-title"
        >
          <h3
            class="font-weight-bold"
          >
            Welcome to Silicon Valley's premier dive cafe!
          </h3>
          <h4>
            Now offering a selection of 
            2
             different snacks and 
            2
             different drinks!
          </h4>
        </div>
      </div>
    </div>
  </section>
</div>
`;
